cards.js

```diff
@@ -16,8 +16,45 @@ function shuffle(array){
   return array;
 }
 
-function buildDeck(){
-
+function buildDeck(values, suits){
 // Put some code in here!
+deck =[]
+for(var v = 0; v < values.length; v++){
+  for(var s = 0; s < suits.length; s++){
+    val = values[v];
+    suit = suits[s];
+      var card = {
+        cardface: val + " of " + suit,
+        value: val,
+        suit: suit,
+        strength: v
+    }
+
+    deck.push(card);
+    }
+  }
+  return deck;
+}
+gameDeck = buildDeck(values, suits);
+  shuffle(gameDeck);
+  for(card in deck){
```

Comment:
For in loops are typically only used for iterating through objects. `deck` here is an array.